satellite/repair/repairer/segments.go

```diff
@@ -78,7 +78,9 @@ func (repairer *SegmentRepairer) Repair(ctx context.Context, path storj.Path) (s
 	// Read the segment pointer from the metainfo
 	pointer, err := repairer.metainfo.Get(ctx, path)
 	if err != nil {
```

Comment:
we need to actually confirm that the error is specifically a not found error. some sort of database connection error here doesn't mean the repair is unnecessary

Reply:
there are some other places in the code where we check if the error is NotFound if that helps

Reply:
I think it would be `if storage.ErrKeyNotFound.Has(err)`

Reply:
Good catch and thank you.